### ISSUE TITLE
resolve_kname, resolve_uname match more than they should

### ISSUE THREAD

**Issue by ajor** [iovisor/bpftrace#174]
Seems to do a prefix search. e.g. searching for `xen_vcpu` actually returns the address of `xen_vcpu_info` as that symbol appears first in my kallsyms:

kallsyms:
```
00000000000140e0 A xen_vcpu_info
0000000000014120 A xen_vcpu_id
0000000000014128 A xen_vcpu
```

```
[alastair@arch build-debug]$ sudo ./src/bpftrace -e 'BEGIN { printf("%p\n", kaddr("xen_vcpu")) }'
Attaching 1 probe...
0x140e0
^C
```

**Comment by cneira** [comment]
@ajor 
I created PR https://github.com/iovisor/bpftrace/pull/195 to fix this.

**Comment by brendangregg** [comment]
fixed by #195
